src/sprintf.js

```diff
@@ -112,10 +113,14 @@
                         arg = (ph.precision ? arg.substring(0, ph.precision) : arg)
                         break
                     case 'x':
-                        arg = (parseInt(arg, 10) >>> 0).toString(16)
-                        break
                     case 'X':
-                        arg = (parseInt(arg, 10) >>> 0).toString(16).toUpperCase()
+                        high = "0"
+                        if (parseInt(arg, 10) > MAXINT - 1 || parseInt(arg, 10) < -MAXINT) {
+                            high = BigInt.asUintN(32, BigInt(arg) >> BigInt(32)).toString(16) // eslint-disable-line
```

Comment:
Had to disable this line because the `eslint` version on this repo does not recognize BigInt, but it is available in all browsers and platforms already: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt